src/main/java/com/minecolonies/coremod/util/TeleportToColony.java

```diff
@@ -0,0 +1,85 @@
+import com.minecolonies.coremod.colony.IColony;
+import com.minecolonies.coremod.commands.MinecoloniesCommand;
+import com.minecolonies.coremod.configuration.Configurations;
+import net.minecraft.command.ICommandSender;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.text.TextComponentString;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Helper class for teleporting players to home or a friendly colony.
+ */
+public final class TeleportToColony
+{
+    private static final String CANT_FIND_COLONY = "No Colony found for teleport, please define one.";
+    private static final String CANT_FIND_PLAYER = "No player found for teleport, please define one.";
+
+    /**
+     * Private constructor to hide the implicit public one.
+     */
+    public TeleportToColony(@NotNull final String... parents)
+    {
+
+    }
+
+    @NotNull
+    public static void ColonyTeleport(@NotNull MinecraftServer server, @NotNull ICommandSender sender, @NotNull String... args){
+        EntityPlayer playerToTeleport;
+        IColony colony;
+        int colonyId;
+        //see if sent by a player and grab their name and Get the players Colony ID that sent the command
+        if (sender instanceof EntityPlayer)
+        {
+            //if no args then this is a home colony TP and we get the players Colony ID
+            if (args[0] == "99999")
+            {
+                playerToTeleport = (EntityPlayer) sender;
+                colony = ColonyManager.getIColonyByOwner(((EntityPlayer) sender).worldObj, (EntityPlayer) sender);
+                colonyId = colony.getID();
+            }
+            else
+            {
+                //if there is args then this will be to a friends colony TP and we use the Colony ID they specify
+                //will need to see if they friendly to destination colony
+                playerToTeleport = (EntityPlayer) sender;
+                colonyId =  Integer.valueOf(args[0]);
+            }
+        }
+        else
+        {
+            sender.getCommandSenderEntity().addChatMessage(new TextComponentString(CANT_FIND_PLAYER));
+            return;
+        }
+            playerToTeleport.getCommandSenderEntity().addChatMessage(new TextComponentString("We got places to go, kid..."));
+            teleportPlayer(playerToTeleport, colonyId);
+    }
+
+    /**
+     * Method used to teleport the player.
+     *
+     * @param colID           the senders colony ID.
+     * @param playerToTeleport the player which shall be teleported.
+     */
+    private static void teleportPlayer(final EntityPlayer playerToTeleport,int colID)
+    {
+        final Colony colony = ColonyManager.getColony(colID);
+        final BlockPos position = colony.getCenter();
+
+        if (colID >= 1)
+        {
+            playerToTeleport.setPositionAndUpdate(position.getX(),position.getY()+2,position.getZ());
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Cast one of the operands of this addition operation to a "double". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS2184)